### PR TITLE
Add Nullable Metric getter to MetricsContainerImpl.

### PR DESCRIPTION
**Pull request by pedapudi**
MetricsContainerImpl currently creates a Counter with a given name when attempting to read that Counter. In order to read potentially non-existent Counters, a getter that doesn't create a Counter on behalf of the reader is necessary. 'tryGetCounter' introduces a way to retrieve null Counter values.

### CODE REVIEW

**Comment by pedapudi** [comment]
@bjchambers Ben, could you please review this?

File: runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java [hunk 1]
```diff
@@ -85,6 +86,9 @@ public CounterCell getCounter(MetricName metricName) {
     return counters.get(metricName);
   }
 
+  @Nullable
+  public CounterCell tryGetCounter(MetricName metricName) { return counters.tryGet(metricName); }
```
Comment (bjchambers) [inline runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java]: Probably worth adding the equivalent for the other get methods (eg., tryGetDistribution).

Comment (bjchambers) [inline runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java]: Javadoc for public  methods?

File: runners/core-java/src/test/java/org/apache/beam/runners/core/metrics/MetricsContainerImplTest.java [hunk 2]
```diff
@@ -67,6 +68,9 @@ public void testCounterDeltas() {
     c1.inc(8L);
     assertThat(container.getUpdates().counterUpdates(), contains(
         metricUpdate("name1", 13L)));
+
+    CounterCell dne = container.tryGetCounter(MetricName.named("ns", "dne"));
+    assertEquals(dne, null);
```
Comment (bjchambers) [inline runners/core-java/src/test/java/org/apache/beam/runners/core/metrics/MetricsContainerImplTest.java]: Probably worth a test that tryGetCounter returns non-null when it does exist?

File: runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java (hunk 1 above)
Comment (pedapudi) [reply runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java]: Added tryGetDistribution and tryGetGauge.

Comment (pedapudi) [reply runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java]: Done.

File: runners/core-java/src/test/java/org/apache/beam/runners/core/metrics/MetricsContainerImplTest.java (hunk 2 above)
Comment (pedapudi) [reply runners/core-java/src/test/java/org/apache/beam/runners/core/metrics/MetricsContainerImplTest.java]: Amended the unit tests.

**Pull request closed without merging**